### PR TITLE
Add a doc page for data mgmt

### PR DESCRIPTION
**Pull request by dmitri-perelman**
Add an overview page for data mgmt of Sui FNs.

## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: doc/src/learn/architecture/data_mgmt.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+The FNs, which are uploading information to the archives, should be configured to ensure that pruning doesn’t occur until the corresponding information has been archived (`use-for-pruning-watermark: true`).
+
+## Full Node key-value store backup
+In order to enable historic data queries for the FNs that prune old transactional data, FN RPC logic is configured to fallback for querying transactional data from a remote store.
+In case the information about the transaction digest, effects, events, or checkpoints is not available locally, a FN retrieves the historical data from a cloud-based key-value store. This fallback retrieval is transparent to the FN clients. Note that the current implementation doesn’t keep historic object versions, and is applied to historic transaction and checkpoint information only. *_We're currently working on a similar setup for retrieving the historic object versions._
+
+
+## Pruning policy examples
```
Comment (dmitri-perelman) [inline doc/src/learn/architecture/data_mgmt.md:68]: @phoenix-o @sadhansood could you please pay special attention to these examples? (just want to verify that they're covering our current guidance and are technically correct).

Comment (phoenix-o) [reply doc/src/learn/architecture/data_mgmt.md:68]: The overall examples look correct, but the current pruning configuration requires more values to be specified. A complete example:
```
authority-store-pruning-config:
  num-latest-epoch-dbs-to-retain: 3
  epoch-db-pruning-period-secs: 3600
  max-checkpoints-in-batch: 10
  max-transactions-in-batch: 1000

  # Prune historic object versions
  num-epochs-to-retain: 0
  # Prune historic transactions of the past epochs
  num-epochs-to-retain-for-checkpoints: 2
  periodic-compaction-threshold-days: 1
```
However, we can make the repeating parameters optional by providing reasonable defaults

Comment (sadhansood) [reply doc/src/learn/architecture/data_mgmt.md:68]: The example makes sense to me.

Comment (dmitri-perelman) [reply doc/src/learn/architecture/data_mgmt.md]: Thanks! Does it look good to you to land now?

**Review by sadhansood — APPROVED** [review]

**Pull request merged**